gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/ConcatenatedGigawordDocuments.java

```diff
@@ -18,18 +26,118 @@
 
 public class ConcatenatedGigawordDocuments implements Iterable<ConcatenatedGigawordDocuments.Article> {
   private final String concatenatedFileText;
+  private List<Article> articleList;
+  boolean annotated;
 
   private ConcatenatedGigawordDocuments(String concatenatedFileText) {
     this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.annotated = false;
   }
 
+  private ConcatenatedGigawordDocuments(String concatenatedFileText, List<Article> aArticleList) {
+    this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.articleList = aArticleList;
+    this.annotated = true;
+  }
+  
   public static ConcatenatedGigawordDocuments fromGigwordGZippedFile(Path p) throws IOException {
     return new ConcatenatedGigawordDocuments(GZIPByteSource.fromCompressed(p.toFile())
             .asCharSource(StandardCharsets.UTF_8).read());
   }
 
+  public static ConcatenatedGigawordDocuments fromAnnotatedGigwordGZippedFile(Path p) throws IOException {
+    // Initialize Streams
+    InputStream fileInputStream = new FileInputStream(p.toString());
```

Comment:
Shouldn't this be done with a try-with-resources block to ensure that the streams are properly closed?